### PR TITLE
Support EdDSA along with ECDSA

### PR DESCRIPTION
**Pull request by chris-wood**
I likely won't have time to generate test vectors for EdDSA today, but we can always add them later.

### CODE REVIEW

File: draft-privacypass-rate-limit-tokens.md [hunk 1]
```diff
@@ -1262,11 +1259,67 @@ unique among all currently in rotation for the origin.
...
+- BKS-BlindPublicKey(pk, bk): Produce a blinded public key based on the input public
+  key pk and blind bk according to {{KEYBLINDING, Section 5.1}}.
+- BKS-Verify(pk, msg, sig): Verify the signature sig over input message msg against
+  the Ed25519 public key pk, as defined in {{RFC8032, Section 5.1.7}}, producing a
+  boolean value indicating success.
+- BKS-BlindKeySign(sk_sign, sk_blind, msg): Sign input message msg with signing key sk_sign and
+  blind sk_blind according to {{KEYBLINDING, Section 5.2}}, yielding an `Nsig=64` byte signature.
+- BKS-SerializePrivatekey(sk): Identity function which outputs sk as an `Nsk=32` byte buffer.
+- BKS-DeserializePrivatekey(buf): Identity function which outputs buf interpreted as `sk`.
+- BKS-SerializePublicKey(pk): Identity function which outputs sk as an `Npk=32` byte buffer.
```
Comment (dvorak42) [inline draft-privacypass-rate-limit-tokens.md:1320]: nit: Should this be output pk?

File: draft-privacypass-rate-limit-tokens.md [hunk 2]
```diff
@@ -1262,11 +1259,67 @@ unique among all currently in rotation for the origin.
 
 This document updates the "Token Type" Registry ({{AUTHSCHEME}}) with the following value:
 
-| Value  | Name                   | Publicly Verifiable | Public Metadata | Private Metadata | Nk  | Nid | Reference        |
-|:-------|:-----------------------|:--------------------|:----------------|:-----------------|:----|:----|:-----------------|
-| 0x0003 | Rate-Limited Blind RSA | Y                   | N               | N                | 512 | 32  | This document    |
+| Value  | Name                                                                 | Publicly Verifiable | Public Metadata | Private Metadata | Nk  | Nid | Reference        |
+|:-------|:---------------------------------------------------------------------|:--------------------|:----------------|:-----------------|:----|:----|:-----------------|
+| 0x0003 | Rate-Limited Blind RSA(SHA-384, 2048-bit) with ECDSA(P-384, SHA-384) | Y                   | N               | N                | 512 | 32  | This document    |
+| 0x0004 | Rate-Limited Blind RSA(SHA-384, 2048-bit) with Ed25519(SHA-512)      | Y                   | N               | N                | 512 | 32  | This document    |
```
Comment (dvorak42) [inline draft-privacypass-rate-limit-tokens.md:1265]: nit: Does the SHA-512 need to be specified here?

**Review by dvorak42 — APPROVED** [review]

File: draft-privacypass-rate-limit-tokens.md (hunk 1 above)
Comment (chris-wood) [inline draft-privacypass-rate-limit-tokens.md:1320]:
```suggestion
- BKS-SerializePublicKey(pk): Identity function which outputs pk as an `Npk=32` byte buffer.
```

File: draft-privacypass-rate-limit-tokens.md (hunk 2 above)
Comment (chris-wood) [reply draft-privacypass-rate-limit-tokens.md:1265]: Yeah, since we use it with HKDF to compute anon_origin_id.

**Review by tfpauly — APPROVED** [review]

**Review by smhendrickson — APPROVED** [review]

**Pull request merged**